src/api/uploads/BaseUpload.js

```diff
@@ -60,11 +74,55 @@ class BaseUpload extends Base {
         this.xhr.options({
             url,
             data: attributes,
-            successHandler: this.preflightSuccessHandler,
+            successHandler: response => {
+                if (this.isUploadFallbackLogicEnabled) {
+                    this.preflightSuccessReachabilityHandler(response);
+                } else {
+                    this.preflightSuccessHandler(response);
+                }
+            },
             errorHandler: this.preflightErrorHandler,
         });
     };
 
+    /**
+     * Handles successful preflight response.
+     * Performs a upload reachability test before calling preflightSuccessHandler.
+     *
+     * @param {Object} - Request options
+     * @return {Promise} Async function promise
+     */
+    preflightSuccessReachabilityHandler = async ({ data }: { data: { upload_url?: string } }): Promise<any> => {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        const uploadUrl = data.upload_url;
+        // If uploadUrl is not available, don't make reachability test
+        if (!uploadUrl) {
+            this.preflightSuccessHandler({ data });
+            return;
+        }
+
+        const uploadHost = this.getUploadHostFromUrl(uploadUrl);
+        // The default upload host should always be reachable
+        if (uploadHost === `${DEFAULT_HOSTNAME_UPLOAD}/`) {
+            this.preflightSuccessHandler({ data });
+            return;
+        }
```

Comment:
Could this be combined with the guard clause above on L102?

Reply:
I don't it can since, L102 checks if the `uploadUrl` is defined, while here we are checking the `uploadHost` which needs to be parsed from the `uploadUrl`